### DISCUSSION TITLE
Sshuttle killer?

### DISCUSSION THREAD

**Discussion by fullmetal-fred** [mudler/edgevpn#91] [Ideas]
Hey Mudler!

I was just thinking about using EdgeVPN as a way to enable access from a local device to a remote network using P2P. The key would be being able to, just-in-time, deploy an EdgeVPN container in a remote network (K3s etc) and have a desktop client (Windows) which can get the same token, set up a tun interface locally, intercept all traffic, and pass it to the remote network. The remote container would need to forward requests to other machines it can route to, and forward DNS requests. This could be used as a just-in-time remote support solution for IoT, AV and other systems where permissive access to a remote network is required.

Just pondering here...

**Comment by mudler** [comment]
Hey! That's actually a cool Idea, I think all the pieces are there, and would just require to configure the host to route all the packets way back to the network/tunnel.

Kairos has the Entangle CRD available to cover further automation with Kubernetes, that I think might help in this setup, docs are here: https://kairos.io/docs/reference/entangle/. It can share services https://kairos.io/docs/reference/entangle/#expose-services and also setup a daemonset for the nodes to access the tunnel: https://github.com/kairos-io/entangle/blob/master/tests/fixtures/vpn.yaml
